tests/test-cross-context-dispatch/ccd-ee8-webapp/src/main/java/org/eclipse/jetty/tests/ccd/ee8/InternalRequestURIFilter.java

```diff
@@ -0,0 +1,55 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.tests.ccd.ee8;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * A servlet filter that will harshly change the return value of
+ * {@link HttpServletRequest#getRequestURI()} to something that does
+ * not satisfy the Servlet spec URI invariant {@code request URI == context path + servlet path + path info}
+ */
+public class InternalRequestURIFilter implements Filter
```

Comment:
Is this attempting to test the object wrapper identity that I mentioned in a subsequent comment? If so I think we need a different test that checks exact object equality.